src/cli/README_SRP.md

```diff
@@ -0,0 +1,181 @@
+
+### Build with SRP Server & Client support
+
+Use the `SRP_SERVER=1 SERVICE=1 ECDSA=1` build switches to enable SRP Server API support, and `SRP_CLIENT=1 ECDSA=1` build switches for SRP Client API support.
+
+Make simulation SRP Server & Client:
+
+```bash
+./bootstrap
+make -f examples/Makefile-simulation SRP_SERVER=1 SERVICE=1 SRP_CLIENT=1 ECDSA=1
+```
+
+## Simulation Example
+
+### Start SRP Server
+
+start the SRP Server node with below command:
+
+```bash
+./output/simulation/bin/ot-cli-ftd 1
+```
+
+Execute below command to setup a Thread network and start the SRP Server:
+
+```bash
+> panid 0xface
+Done
+> channel 18
+Done
+> masterkey 00112233445566778899aabbccddeeff
+Done
+> ifconfig up
+Done
+> thread start
+Done
+> state
+leader
+Done
+>ipaddr
+fd00:db8:0:0:0:ff:fe00:fc10
+fd00:db8:0:0:0:ff:fe00:fc00
+fd00:db8:0:0:0:ff:fe00:8400
+fd00:db8:0:0:7cdf:b15e:30e0:df4e
+fe80:0:0:0:146e:a00:0:1
+Done
+> srp server enable
+Done
+> netdata show
+Prefixes:
+Routes:
+Services:
+44970 5d c002 s 8400
+Done
+```
+
+The SRP Server listening UDP port which is `c002`(`49154`) is included in the Server Data (listed by the `netdata show` command).
+
+### Start SRP Client
+
+start the SRP Client node with below command:
+
+```bash
+./output/simulation/bin/ot-cli-ftd 2
+```
+
+Type below command to join the Thread Network and register an `_ipps._tcp` service:
+
+```bash
+> panid 0xface
+Done
+> channel 18
+Done
+> masterkey 00112233445566778899aabbccddeeff
+Done
+> ifconfig up
+Done
+> thread start
+Done
+> state
+child
+Done
+> ipaddr
+fd00:db8:0:0:0:ff:fe00:8401
+fd00:db8:0:0:2433:96cd:5243:c544
+fe80:0:0:0:146e:a00:0:2
+Done
+> srp client host name my-host
+Done
+> srp client host address fd00:db8:0:0:2433:96cd:5243:c544
+Done
+> srp client service add my-service _ipps._tcp 12345
+Done
+srp client start fd00:db8:0:0:7cdf:b15e:30e0:df4e 49154
+Done
+```
+
+Make sure the SRP Server address & port are used for the `srp client start` command.
+
+### Verify the service status
+
+Check if the host & service has been successfully registered on the client node:
+
+```bash
+> srp client host
+name:"my-host", state:Registered, addrs:[fd00:db8:0:0:2433:96cd:5243:c544]
+Done
+> srp client service
+instance:"my-service", name:"_ipps._tcp", state:Registered, port:12345, priority:0, weight:0
+Done
+```
+
+Make sure it shows `state:Registered` for both host & service commands.
+
+Check the host & service on the server node:
+
+```bash
+> srp server host
+my-host.default.service.arpa.
+    deleted: false
+    addresses: [fd00:db8:0:0:2433:96cd:5243:c544]
+Done
+> srp server service
+my-service._ipps._tcp.default.service.arpa.
+    deleted: false
+    port: 12345
+    priority: 0
+    weight: 0
+    TXT: 00
+    host: my-host.default.service.arpa.
+    addresses: [fd00:db8:0:0:2433:96cd:5243:c544]
+Done
+```
+
+Make sure it shows `deleted: false` for both host & service commands.
+
+### Remove the service
+
+Use below command on the client node to remove the service:
```

Comment:
A service can be removed through the client node:

(I assume this is the only way to remove)